### PR TITLE
Garden additions

### PR DESCRIPTION
**Pull request by Procyonae**
#### Summary
Content "Adds more garden items/furniture/terrain"

#### Purpose of change

Gardens sad

#### Describe the solution

Adds functional push lawnmower items and lawn terrain using a new GRASS_CUT_FINE quality
Adds trellis fence and archways
Adds plastic and rattan furniture with recipes
Adds beach and patio umbrellas with table variants to fit them
Adds a dartboard and darts (I appreciate dartboards aren't really related they just don't seem worthy of a separate PR)
Adds small stone, metal and ceramic/resin statues for things like flamingos and gnomes ^-^
Adds a hook to hang a windchime, wooden windchime or functional hanging basket on
Adds patio tiles and deck board terrains
Adds a disposable barbecue
Adds pruning shears and loppers
Adds sculpted hedge and hedge archways and constructions to cut/sculpt hedges
Adds constructible lined pond

Cleans up the construction menu a smidgen to make up for the bloat I'm adding

#### Describe alternatives you've considered

Sticking to map stuff

Adding an art skill and letting you craft the statues :^)

Making trellis act like planters
Adding a construction for patio tiles

Hardcoding the gas lawnmower use action so it can fail like other gas powered things

Other cool additions if anyone wants a crack:
Solar powered night lights
A mini standing greenhouse

#### Additional context

I still plan on finishing this as of April 2024 I just forget what needs doing to complete it and reviewing the whole thing to work it out and dealing with the merge conflicts will be pain so it's on the back burner

### CODE REVIEW

**Comment by Karol1223** [comment]
Would it be alright to ask of you to give the new items/ones you reworked the new price and to-hit value ways to handle things? You're manually defining to-hit instead of using the modifiers and you can define price specifically in USD instead of just raw numbers. If you're not okay with it I can do it later on (or probably I could do it via reviews here and you'd just merge it afterwards if you'd rather have it like this)

**Comment by Procyonae** [comment]
> Would it be alright to ask of you to give the new items/ones you reworked the new price and to-hit value ways to handle things? You're manually defining to-hit instead of using the modifiers and you can define price specifically in USD instead of just raw numbers. If you're not okay with it I can do it later on (or probably I could do it via reviews here and you'd just merge it afterwards if you'd rather have it like this)

Sure

**Comment by ashGlaw** [comment]
The solar-powered night lights are a fantastic idea! Especially if they give a solar cell on disassembly.

File: data/json/furniture_and_terrain/furniture-decorative.json [hunk 1]
```diff
@@ -155,6 +155,28 @@
...
+    "flags": [ "MINEABLE", "TRANSPARENT" ],
+    "bash": {
+      "str_min": 16,
+      "str_max": 40,
+      "sound": "smash!",
+      "sound_fail": "thump.",
+      "items": [
+        { "item": "rock", "count": [ 1, 6 ] },
+        { "item": "hose", "count": [ 0, 1 ] },
+        { "item": "copper", "count": [ 0, 5 ] }
```
Comment (TheShadowFerret) [inline data/json/furniture_and_terrain/furniture-decorative.json:176]: Why is there copper in this 🤔

Comment (Procyonae) [reply data/json/furniture_and_terrain/furniture-decorative.json:176]: The pump, idk it's very first pass rn

File: data/json/items/tool/landscaping.json [hunk 2]
```diff
@@ -180,6 +180,139 @@
...
+    "to_hit": { "grip": "none", "length": "long", "surface": "line", "balance": "clumsy" },
+    "melee_damage": { "bash": 6 }
+  },
+  {
+    "id": "lawnmower_on",
+    "copy-from": "lawnmower",
+    "type": "TOOL",
+    "symbol": ";",
+    "color": "green",
+    "name": { "str": "lawnmower (on)" },
```
Comment (Procyonae) [inline data/json/items/tool/landscaping.json:246]:
```suggestion
    "name": { "str": "lawnmower (on)", "str_pl": "lawnmowers (on)" },
```

File: data/json/items/resources/plastic.json [hunk 3]
```diff
@@ -139,6 +139,24 @@
...
     "flags": [ "NO_SALVAGE" ]
   },
+  {
+    "id": "pond_liner",
+    "type": "GENERIC",
+    "category": "spare_parts",
+    "name": { "str": "pond liner" },
+    "description": "A large sheet of thick but flexible plastic, the sort that would have been used for sealing a pond.",
+    "weight": "851 g",
+    "volume": "31 ml",
```
Comment (Karol1223) [inline data/json/items/resources/plastic.json:149]: Can't tell you how you calculated these numbers but this is denser than tungsten

**Review by Karol1223 — CHANGES_REQUESTED** [review]

File: data/json/items/resources/plastic.json (hunk 3 above)
Comment (Procyonae) [reply data/json/items/resources/plastic.json]: Idek how I managed that ;D

File: data/json/items/containers.json [hunk 4]
```diff
@@ -122,6 +122,25 @@
...
     "melee_damage": { "bash": 8 }
   },
+  {
+    "id": "bag_lawnmower",
+    "type": "GENERIC",
+    "category": "container",
+    "name": { "str": "lawnmower bag" },
+    "looks_like": "bag_canvas",
+    "description": "A large and sturdy canvas sack.  Smells faintly of cut grass.",
+    "//": "Based off of https://www.amazon.com/964-04117A-664-04117A-Bag-Compatible-964-04117B/dp/B097MNCDHQ/",
```
Comment (Procyonae) [inline data/json/items/containers.json]:
```suggestion
    "//": "Based off of https://www.amazon.com/dp/B097MNCDHQ/",
```

File: data/json/items/fluff.json [hunk 5]
```diff
@@ -546,6 +546,21 @@
     "flags": [ "NO_REPAIR" ],
     "use_action": [ "PLAY_GAME" ]
   },
+  {
+    "id": "dartboard",
+    "type": "GENERIC",
+    "category": "other",
+    "name": { "str": "dartboard" },
+    "description": "A game where players throw darts at the board trying to score exact values.",
+    "//": "Based off https://www.amazon.com/Franklin-Sports-Regulation-Bristle-Dartboard/dp/B0091DKL4W/",
```
Comment (Procyonae) [inline data/json/items/fluff.json:555]:
```suggestion
    "//": "Based off https://www.amazon.com/dp/B0091DKL4W/",
```

File: data/json/items/ranged/throwing.json [hunk 6]
```diff
@@ -65,6 +65,25 @@
...
+    "type": "GENERIC",
+    "id": "dart",
+    "category": "weapons",
+    "price": 55,
+    "price_postapoc": 10,
+    "name": { "str": "dart" },
+    "symbol": ";",
+    "color": "black",
+    "description": "A dart designed for playing darts.",
+    "//": "Based off https://www.amazon.co.uk/CENTAUR-Professional-Frosted-Shafts-Flights/dp/B0BFQ86NC7/ , darts seem to range from 18-30g so I took the average",
```
Comment (Procyonae) [inline data/json/items/ranged/throwing.json:78]:
```suggestion
    "//": "Based off https://www.amazon.co.uk/dp/B0BFQ86NC7/ , darts seem to range from 18-30g so I took the average",
```

File: data/json/items/tool/cooking.json [hunk 7]
```diff
@@ -226,6 +226,54 @@
     "use_action": [ "HOTPLATE" ],
     "melee_damage": { "bash": 4 }
   },
+  {
+    "id": "charcoal_cooker_disposable",
+    "type": "TOOL",
+    "name": { "str": "disposable barbecue" },
+    "description": "A small metal tray for cooking food while camping, only good for one use.  Lasts about 40 minutes.",
+    "//": "Based off https://www.amazon.com/Improved-Disposable-Size-Charcoal-Camping-Tailgate/dp/B087YHNJ2M/ , lasts 45 mins according to review",
```
Comment (Procyonae) [inline data/json/items/tool/cooking.json:234]:
```suggestion
    "//": "Based off https://www.amazon.com/dp/B087YHNJ2M/ , lasts 45 mins according to review",
```

File: data/json/items/tool/landscaping.json [hunk 8]
```diff
@@ -180,6 +180,138 @@
...
   },
+  {
+    "id": "lawnmower_reel",
+    "type": "TOOL",
+    "symbol": ";",
+    "color": "light_gray",
+    "looks_like": "lawnmower",
+    "name": { "str": "reel lawnmower" },
+    "description": "A hand powered reel pushmower.",
+    "//": "Based off https://www.amazon.com/American-Lawn-Mower-Company-1204-14/dp/B00004RA3F/ , used a more sane depth of 12in",
```
Comment (Procyonae) [inline data/json/items/tool/landscaping.json:191]:
```suggestion
    "//": "Based off https://www.amazon.com/dp/B00004RA3F/ , used a more sane depth of 12in",
```

File: data/json/items/tool/landscaping.json [hunk 9]
```diff
@@ -180,6 +180,138 @@
...
+    "id": "lawnmower",
+    "type": "TOOL",
+    "symbol": ";",
+    "color": "red",
+    "name": { "str": "lawnmower (off)", "str_pl": "lawnmowers (off)" },
+    "description": "A motorized pushmower that takes gasoline as fuel.",
+    "price": "220 USD",
+    "price_postapoc": 50,
+    "material": [ "steel", "plastic" ],
+    "//": "Based off https://www.amazon.com/PowerSmart-Powered-4-Stroke-Powerful-Included/dp/B0BWJB4BZ3/ , uses electric's volume and length for consistency",
```
Comment (Procyonae) [inline data/json/items/tool/landscaping.json:212]:
```suggestion
    "//": "Based off https://www.amazon.com/dp/B0BWJB4BZ3/ , uses electric's volume and length for consistency",
```

File: data/json/items/tool/landscaping.json [hunk 10]
```diff
@@ -180,6 +180,138 @@
...
+    "type": "TOOL",
+    "symbol": ";",
+    "color": "green",
+    "looks_like": "lawnmower",
+    "name": { "str": "electric lawnmower (off)", "str_pl": "electric lawnmowers (off)" },
+    "description": "A cordless electric motorized pushmower.  You could take it apart if you had a wrench.",
+    "price": "150 USD",
+    "price_postapoc": 70,
+    "material": [ "steel", "plastic" ],
+    "//": "Based off https://www.amazon.com/Greenworks-Cordless-Brushless-Included-LMF402/dp/B07MJTJJ5X/",
```
Comment (Procyonae) [inline data/json/items/tool/landscaping.json:269]:
```suggestion
    "//": "Based off https://www.amazon.com/dp/B07MJTJJ5X/",
```

File: data/json/items/tool/landscaping.json [hunk 11]
```diff
@@ -298,6 +430,46 @@
     "weapon_category": [ "SHORT_SWORDS" ],
     "melee_damage": { "bash": 3, "cut": 12 }
   },
+  {
+    "id": "pruning_shears",
+    "type": "TOOL",
+    "name": { "str_sp": "pruning shears" },
+    "description": "A gardening tool used for trimming plants.  While sharp, they are incredibly awkward to use for anything but their intended purpose.",
+    "//": "Based off of https://www.amazon.com/Fiskars-91095935J-Bypass-Pruning-Shears/dp/B00002N66H/ Weapon stats treated as though you're stabbing with them open.  Butchering 3 taken from scissors",
```
Comment (Procyonae) [inline data/json/items/tool/landscaping.json:438]:
```suggestion
    "//": "Based off of https://www.amazon.com/dp/B00002N66H/ Weapon stats treated as though you're stabbing with them open.  Butchering 3 taken from scissors",
```

File: data/json/items/tool/landscaping.json [hunk 12]
```diff
@@ -298,6 +430,46 @@
...
+    "techniques": [ "WBLOCK_1" ],
+    "flags": [ "BELT_CLIP", "DURABLE_MELEE" ],
+    "melee_damage": { "bash": 1, "stab": 5 }
+  },
+  {
+    "id": "loppers",
+    "type": "TOOL",
+    "name": { "str_sp": "loppers" },
+    "description": "A gardening tool used for cutting tree branches and the like.  While sharp, they are incredibly awkward to use for anything but their intended purpose.",
+    "//": "Based off of https://www.amazon.com/Fiskars-Power-Lever-Bypass-Lopper-28-Inch/dp/B007HZCR02/ Weapon stats treated as though you're bashing with them.  Butchering 3 taken from scissors",
```
Comment (Procyonae) [inline data/json/items/tool/landscaping.json:458]:
```suggestion
    "//": "Based off of https://www.amazon.com/dp/B007HZCR02/ Weapon stats treated as though you're bashing with them.  Butchering 3 taken from scissors",
```

File: data/json/items/tool/misc.json [hunk 13]
```diff
@@ -885,13 +885,54 @@
...
     "flags": [ "RAIN_PROTECT", "SHEATH_SWORD", "FRAGILE_MELEE" ],
     "weapon_category": [ "FENCING_WEAPONRY" ],
     "melee_damage": { "bash": 6, "stab": 4 }
   },
+  {
+    "id": "umbrella_patio",
+    "type": "GENERIC",
+    "name": { "str": "patio umbrella" },
+    "description": "A large umbrella designed to fit a table.",
+    "//": "Based on https://www.amazon.com/SUNNYARD-Umbrella-Market-Outdoor-Garden/dp/B079Q84RY5/",
```
Comment (Procyonae) [inline data/json/items/tool/misc.json]:
```suggestion
    "//": "Based on https://www.amazon.com/dp/B079Q84RY5/",
```

File: data/json/items/tool/misc.json [hunk 14]
```diff
@@ -885,13 +885,54 @@
...
+    "techniques": [ "WBLOCK_1" ],
+    "flags": [ "RAIN_PROTECT", "POLEARM", "REACH_ATTACK", "REACH3" ],
+    "melee_damage": { "bash": 12 }
+  },
+  {
+    "id": "umbrella_beach",
+    "type": "GENERIC",
+    "name": { "str": "beach umbrella" },
+    "description": "A large umbrella designed to stick in the sand or fit a table.",
+    "//": "Based on https://www.amazon.com/Clothclose-Beach-Umbrella-for-Sand/dp/B0BNX5KCMX/",
```
Comment (Procyonae) [inline data/json/items/tool/misc.json:920]:
```suggestion
    "//": "Based on https://www.amazon.com/dp/B0BNX5KCMX/",
```

**Review by Procyonae — COMMENTED** [review]
Realised you can shorten amazon links more than I have been

**Comment by ilovehans10** [comment]
I am almost definite that this is outside the scope of this pr, but having wind chimes that make sounds when it is windy would be so cool. More realistically, would it be possible to make it so that the wind chimes item can be played to make some noise or possibly even make noise with the furniture version.

I see that wall hooks are included in this PR, but would something like a [shepherds hook](https://www.amazon.com/Derkniel-Shepherd-Heavy-Duty-Hanging-Baskets/dp/B0991TDCF8) be possible as well?

I also like the darts idea and was wondering if making it so that they can be used to train throwing would be something that should be a different PR or if that could be snuck in here.

**Comment by Procyonae** [comment]
I'm not adding anything else to this PR it's just looking for me to bother finishing it but I like all those ideas, I don't really fancy diving into how to make the windchime functional but a shepard's hook shouldn't cause any problems the hook doesn't (and I haven't found any). Equally idk why I haven't added bird/beehouses.
A practice throwing recipe with the dartboard makes alot of sense but I'll leave it until I add spawns (this PR literally adds 0 spawns, I'm hoping it's considered mergeable based on trust that I'll bother adding them later pretty much)

**Pull request closed without merging**